src/main/java/com/denizenscript/denizencore/tags/core/ProcedureScriptTagBase.java

```diff
@@ -1,90 +1,93 @@
-            if (attribute.getParam().indexOf('.') > 0) {
+        TagManager.registerTagHandler(ObjectTag.class, ElementTag.class, "proc", (attribute, object) -> {
+            ScriptTag script;
+            String path = null;
+            if (object.asString().indexOf('.') > 0) {
                 String[] split = attribute.getParam().split("\\.", 2);
                 path = split[1];
                 script = ScriptTag.valueOf(split[0], attribute.context);
             }
             else {
                 script = attribute.paramAsType(ScriptTag.class);
             }
-        }
-        else {
-            Debug.echoError("Invalid procedure script tag!");
-            return;
-        }
-        if (script == null) {
-            attribute.echoError("Missing script for procedure script tag '" + attribute.getParam() + "'!");
-            return;
-        }
-        if (!(script.getContainer() instanceof ProcedureScriptContainer)) {
-            attribute.echoError("Chosen script is not a procedure script!");
-            return;
-        }
-        ListTag definitions = null;
+            if (script == null) {
+                attribute.echoError("Missing script for procedure script tag '" + attribute.getParam() + "'!");
+                return null;
+            }
+            if (!(script.getContainer() instanceof ProcedureScriptContainer)) {
+                attribute.echoError("Chosen script is not a procedure script!");
+                return null;
+            }
+            ListTag definitions = null;
+            MapTag mappedDefinitions;
 
-        // <--[tag]
-        // @attribute <proc[<procedure_script_name>].context[<object>|...]>
-        // @returns ObjectTag
-        // @description
-        // Returns the 'determine' result of a procedure script with the given context.
-        // -->
-        if (attribute.startsWith("context", 2)) {
-            definitions = attribute.contextAsType(2, ListTag.class);
+            // <--[tag]
+            // @attribute <proc[<procedure_script_name>].context[<object>|...]>
+            // @returns ObjectTag
+            // @description
+            // Returns the 'determine' result of a procedure script with the given context.
+            // -->
+            if (attribute.startsWith("context", 2)) {
+                mappedDefinitions = null;
+                definitions = attribute.contextAsType(2, ListTag.class);
+                attribute.fulfill(1);
+            }
+
+            // <--[tag]
+            // @attribute <proc[<procedure_script_name>].context_map[<map>]>
+            // @returns ObjectTag
+            // @description
+            // Returns the 'determine' result of a procedure script with the given context.
+            // -->
+            else if (attribute.startsWith("context_map", 2)) {
+                mappedDefinitions = attribute.contextAsType(2, MapTag.class);
+                attribute.fulfill(1);
+            } else {
+                mappedDefinitions = null;

```

Comment:
having this explicit `mappedDefinitions = null` spam is redundant, just preset to null in the declaration